### PR TITLE
[B2BP-216] - Terraform IAM role deploy CMS and Website AWS

### PR DESCRIPTION
**Pull request by maurodandrea**
Script Terraform for create IAM Role necessary for deploy of CMS and Website on AWS

#### List of Changes
Add Script Terraform IAM Role

#### Motivation and Context
Script Terraform for create IAM Role necessary for deploy of CMS and Website on AWS

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change not requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .infrastructure/09_iam_role_deploy.tf [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "deploy_ecs" {
+  name        = "PagoPaECSDeploy"
+  description = "Policy to allow deploy on ECS."
+
+  policy = jsonencode(
```
Comment (datalek) [inline .infrastructure/09_iam_role_deploy.tf:114]: Can you use the same syntax as [this one](https://github.com/pagopa/b2b-portals/pull/58/files#diff-f9af473ca5e59e9c0f9867018d548d1eb8c2fa0183ddc85d83fa222870a493e8R10-R30)?

File: .infrastructure/09_iam_role_deploy.tf [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            "token.actions.githubusercontent.com:aud" : "sts.amazonaws.com"
+          }
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "deploy_ecs" {
+  name        = "PagoPaECSDeploy"
```
Comment (datalek) [inline .infrastructure/09_iam_role_deploy.tf:111]:
```suggestion
  name        = "DeployECS"
```

**Review by datalek — CHANGES_REQUESTED** [review]

File: .infrastructure/09_iam_role_deploy.tf (hunk 1 above)
Comment (maurodandrea) [reply .infrastructure/09_iam_role_deploy.tf:114]: the syntax is already the same for the two roles, they are defined identically. "policy = jsonencode" is used for policy, "assume_role_policy = jsonencode" is for role. Or maybe I didn't understand the observation well ?

Comment (datalek) [reply .infrastructure/09_iam_role_deploy.tf:114]: e.g.: one is defined as `"Version" : "2012-10-17"` and other as `Version = "2012-10-17"`

File: .infrastructure/09_iam_role_deploy.tf [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+###############################################################################
+#                      IAM Role to use on deploy CMS                       #
+###############################################################################
+resource "aws_iam_role" "deploy_ecs" {
+  name        = "GitHubActionDeployECS"
+  description = "Role to assume to deploy on ECS."
+
+
+  assume_role_policy = jsonencode({
```
Comment (datalek) [inline .infrastructure/09_iam_role_deploy.tf:87]: This policy is the same as [this one](https://github.com/pagopa/b2b-portals/pull/58/files#diff-f9af473ca5e59e9c0f9867018d548d1eb8c2fa0183ddc85d83fa222870a493e8R10-R30), can we remove the duplication?

Comment (maurodandrea) [reply .infrastructure/09_iam_role_deploy.tf:87]: The policy is defined with JSON within each role. I create it externally as a policy and link it to both roles, okay?

File: .infrastructure/09_iam_role_deploy.tf (hunk 1 above)
Comment (maurodandrea) [reply .infrastructure/09_iam_role_deploy.tf]: change present in the latest commit

File: .infrastructure/09_iam_role_deploy.tf (hunk 3 above)
Comment (maurodandrea) [reply .infrastructure/09_iam_role_deploy.tf]: change present in the latest commit

File: .infrastructure/09_iam_role_deploy.tf [hunk 4]
```diff
@@ -0,0 +1,130 @@
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "deploy_github" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+    principals {
+      type        = "Federated"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/token.actions.githubusercontent.com"]
+    }
```
Comment (kin0992) [inline .infrastructure/09_iam_role_deploy.tf:10]:
```suggestion
    }
    
```

**Review by kin0992 — APPROVED** [review]

**Review by kin0992 — APPROVED** [review]

**Pull request merged**